src/App.jsx

```diff
@@ -19,20 +21,25 @@ export function App() {
 	 * to create and join a new list.
 	 */
 	const [listToken, setListToken] = useStateWithStorage(
-		'my test list',
+		null,
 		'tcl-shopping-list-token',
 	);
 
+	const createNewList = () => {
+		const newToken = generateToken();
+		setListToken(newToken);
+	};
+
```

Comment:
If this function is being passed as a prop, I would highly suggest utilizing the [`React.useCallback` hook](https://reactjs.org/docs/hooks-reference.html#usecallback). `useCallback` creates a `stable reference` to the function, so there are no "accidental re-renders" of the `Home` component. 

I found a [good video](https://egghead.io/lessons/react-avoiding-accidental-re-renders-with-the-usecallback-hook) that shows issues that might arise when you don't wrap this method in `useCallback`.

Reply:
@jendevelops thank you for the feedback. I've never used `React.useCallback`, so it is great to get some exposure to it. If I'm understanding correctly, you are suggesting that `useCallback` is used in `Home.jsx`. Something like:
```
const handleClick = useCallback(() => {
		createNewList();
	}, [createNewList]);
```

The video was helpful, and the guy's keyboard sounds like a midwestern rain shower, so that's nice ( =

Reply:
You would actually be wrapping the whole method like this
```suggestion
	const createNewList = useCallback(() => {
		const newToken = generateToken();
		setListToken(newToken);
	}, [setListToken]);

```

This way `createNewList` is a function that is "protected" by `useCallback`, preventing new references being created on re-renders not triggered with a change to `setListToken`.

Reply:
Well, I'm glad I checked! Thanks for the clear explanation; I understand now!